### PR TITLE
deploy pgweb for postgresql web interface

### PR DESCRIPTION
**Pull request by harshad16**
deploy pgweb for postgresql web interface
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

### CODE REVIEW

File: core/overlays/test/pgweb.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          httpGet:
+            path: '/'
+            port: 8081
+            scheme: HTTP
+          initialDelaySeconds: 10
+          periodSeconds: 60
+          timeoutSeconds: 10
+  test: false
+  triggers:
+  - type: ConfigChange
```
Comment (goern) [inline core/overlays/test/pgweb.yaml:81]: configChange trigger is a little bit tricky, lets see how it behaves...

File: core/overlays/test/pgweb.yaml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+kind: DeploymentConfig
+metadata:
+  labels:
+    service: pgweb
+    app: thoth
+    component: pgweb
+  name: pgweb
+spec:
+  replicas: 1
+  selector:
```
Comment (goern) [inline core/overlays/test/pgweb.yaml:25]: https://github.com/thoth-station/thoth-application/blob/95d6dea7e6015c5e37dd9740c0f72695f64d5a4f/core/overlays/test/kustomization.yaml#L3 will add more labels, so the selector and the template labels need to include them. feel free to remove the version commonLabel ;)

File: core/overlays/test/pgweb.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        - pgweb
+      from:
+        kind: ImageStreamTag
+        name: 'pgweb:latest'
+---
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: pgweb
+  labels:
```
Comment (goern) [inline core/overlays/test/pgweb.yaml:95]: we dont need these

File: core/overlays/test/pgweb.yaml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    component: pgweb
+spec:
+  to:
+    kind: Service
+    name: pgweb
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
```
Comment (goern) [inline core/overlays/test/pgweb.yaml:106]: see other comments on commonLabel

File: core/overlays/test/pgweb.yaml (hunk 1 above)
Comment (harshad16) [reply core/overlays/test/pgweb.yaml]: i have removed this for now, as we should first fix this in amun-api or user-api, then we can replicate the behavior in other deployment configs.

File: core/overlays/test/pgweb.yaml (hunk 2 above)
Comment (harshad16) [reply core/overlays/test/pgweb.yaml:23]: Added the commonlabel to service's selector and template's label.

**Comment by goern** [comment]
Do we really need this in stage, or can we life w/o it?

@fridex @pacospace

**Comment by fridex** [comment]
> Do we really need this in stage, or can we life w/o it?
> 
> @fridex @pacospace

I'm fine without having it. We have periodic dumps that can be downloaded, no need to expose stage database this way - users should not have it accessible.

**Comment by harshad16** [comment]
we do want it for the test though right?

**Comment by fridex** [comment]
> we do want it for the test though right?

Yes, let's keep it for the test environment.

**Pull request closed without merging**